kythe/proto/repo.proto

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2018 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+syntax = "proto3";
+
+package kythe.proto.repo;
+
+// Represents configuration data on how to extract a repo.
+message Config {
+  repeated ExtractionHints extraction = 1;
+}
+
+// Settings specific to extracting a single root and build system in this repo.
+message ExtractionHints {
+  BuildSystem build_system = 1;
+  // The root of this extraction target, relative to the repo's root.
+  // The build config (e.g. pom.xml) should live in this directory.
+  string root = 2;
+
+  JavaOptions java_options = 3;
+}
+
+enum BuildSystem {
+  UNKNOWN = 0;
+
+  MAVEN = 1;
+  GRADLE = 2;
+  BAZEL = 3;
+}
+
+message JavaOptions {
+  enum Version {
```

Comment:
Making this an enum is interesting. I understand the motivation for it, and will strive to be optimistic that it sticks this way for all the versions we need to identify.

Reply:
Yup, and we can always change it later.